### PR TITLE
Added new test case for storage manager.

### PR DESCRIPTION
**Pull request by IcedCoffeeBoy**

### CODE REVIEW

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 1]
```diff
@@ -58,6 +62,21 @@ public void prefsReadSave() throws Exception {
...
     }
 
+    @Test
+    public void handleAddressBookAndEventListChangedEvent_exceptionThrown_eventRaised() {
+        // Create a StorageManager while injecting a stub that  throws an exception when the save method is called
+        Storage storage = new StorageManager(new XmlAddressBookStorageExceptionThrowingStub(Paths.get("dummy")),
+                new XmlEventStorageExceptionThrowingStub(Paths.get("dummy")),
+                new JsonUserPrefsStorage(Paths.get("dummy")));
+        storage.handleAddressBookChangedEvent(new AddressBookChangedEvent(new AddressBook()));
+        assertTrue(eventsCollectorRule.eventsCollector.getMostRecent() instanceof DataSavingExceptionEvent);
```
Comment (jieliangang) [inline src/test/java/seedu/address/storage/StorageManagerTest.java]: would it be better to separate this into two different tests? for addressbook and eventlist separately

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 2]
```diff
@@ -102,6 +111,27 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) thro
         }
     }
 
+
+    // ================ EventList tests ==============================
+    @Test
+    public void eventListReadSave() throws Exception {
+        /*
+         * Note: This is an integration test that verifies the StorageManager is properly wired to the
+         * {@link XmlAddressBookStorage} class.
```
Comment (jieliangang) [inline src/test/java/seedu/address/storage/StorageManagerTest.java]: do u mean XmlEventListStorage?

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 3]
```diff
@@ -102,6 +111,27 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) thro
...
+         * More extensive testing of UserPref saving/reading is done in {@link XmlAddressBookStorageTest} class.
```
Comment (jieliangang) [inline src/test/java/seedu/address/storage/StorageManagerTest.java]: same regarding {@link XmlAddressBookStorageTest}

File: src/main/java/seedu/address/model/event/Event.java [hunk 4]
```diff
@@ -89,6 +89,28 @@ public boolean isSameEvent(Event event) {
                 || event.getEndTime().equals(getEndTime()));
     }
 
+    /**
+     * Returns true if both events have the same identity and data fields.
+     * This defines a stronger notion of equality between two events.
+     */
+    @Override
```
Comment (jieliangang) [inline src/main/java/seedu/address/model/event/Event.java]: nice work for the addition of this

**Comment by IcedCoffeeBoy** [comment]
would like to get this pass

**Pull request merged**